Add Memory Estimation Tracker

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #165318
* __->__ #165059
* #164945
* #164944
* #164783
* #164738

Add Memory Tracker utility, which will track live memory given alternate ordering of nodes. 

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben

eellison committed:
Add Memory Estimation Tracker

[ghstack-poisoned]

eellison committed:
Update on "Add Memory Estimation Tracker"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

eellison committed:
Update on "Add Memory Estimation Tracker"


Add Memory Tracker utility, which will track live memory given alternate ordering of nodes. 

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

ezyang reviewed (approved):
I'd also think about making sure this can be used outside of Inductor

eellison committed:
Update on "Add Memory Estimation Tracker"


Add Memory Tracker utility, which will track live memory given alternate ordering of nodes. 

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

eellison commented:
@ezyang there is nothing preventing using this outside inductor.

eellison committed:
Update on "Add Memory Estimation Tracker"


Add Memory Tracker utility, which will track live memory given alternate ordering of nodes. 

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

eellison commented:
@pytorchbot merge
